webapp/Connector/src/app/store/MAb.js

```diff
@@ -0,0 +1,179 @@
+
+            Ext.each(this.mabMixStudies, function(mab) {
+                var stdName = mab.mab_mix_name_std;
+                if (mabStudies[stdName] == null)
+                    mabStudies[stdName] = [];
+                mabStudies[stdName].push(mab);
+            });
+
+            Ext.iterate(mabMixes, function(mixStd){
+                var mixes =  mabMixes[mixStd];
+                var mstudies = mabStudies[mixStd];
+
+                var metaMix = mixes[0];
+                var mixRec = {
+                    mab_mix_name_std: metaMix.mab_mix_name_std,
+                    mab_mix_type: metaMix.mab_mix_type,
+                    mab_mix_lanlid: metaMix.mab_mix_lanlid,
+                    mab_mix_lanlid_link: this.getLanlLink(metaMix.mab_mix_lanlid),
+                    mab_mix_name_other: metaMix.mab_mix_name_other,
+                    other_labels: metaMix.mix_labels
+                };
+
+                var donors = [], isotypes = [], hxb2Locs = [], mabnames = [], mabs, mabIdMap = {};
+
+                // process mix and mab metadata
+                Ext.each(mixes, function(mix) {
+                    var donorSpecies = mix.mab_donor_species;
+                    var isotype = mix.mab_isotype;
+                    var hxb2Loc = mix.mab_hxb2_location;
+                    var mabName = mix.mab_name_std;
+                    if (donorSpecies && donors.indexOf(donorSpecies) === -1)
+                        donors.push(donorSpecies);
+                    if (isotype && isotypes.indexOf(isotype) === -1)
+                        isotypes.push(isotype);
+                    if (hxb2Loc && hxb2Locs.indexOf(hxb2Loc) === -1)
+                        hxb2Locs.push(hxb2Loc);
+                    if (mabName && mabnames.indexOf(mabName) === -1)
+                        mabnames.push(mabName);
+
+                    var mabId = mix.mab_id;
+                    if (mabIdMap[mabId])
+                        return;
+
+                    mabIdMap[mabId] = {
+                        mabName: mabName,
+                        isotype: isotype,
+                        hxb2Loc: hxb2Loc,
+                        mab_lanlid: mix.mab_lanlid,
+                        mab_lanlid_link: this.getLanlLink(mix.mab_lanlid),
+                        bindingType: mix.mab_ab_binding_type,
+                        donorSpecies: donorSpecies,
+                        donorId: mix.mab_donorid,
+                        donorClade: mix.mab_donor_clade
+                    };
+                }, this);
+
+                mixRec.donors = donors;
+                mixRec.donors_str = donors.join(', ');
+                mixRec.isotypes = isotypes;
+                mixRec.isotypes_str = isotypes.join(', ');
+                mixRec.hxb2Locs = hxb2Locs;
+                mixRec.hxb2Locs_str = hxb2Locs.join(', ');
+                mixRec.mabnames_str = mabnames.join(', ');
+
+                mabs = Object.values(mabIdMap);
+                mabs.sort(function(a, b) {
```

Comment:
We should use `Ext.Object.getKeys` here in lieu of `Object.values` since [IE does not support Object.values](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_objects/Object/values#Browser_compatibility).

```js
mabs = Ext.Object.getKeys(mapIdMap).sort(/* ... */);
```

Should probably scan the app for other usages as well.